daffodil-runtime2/src/test/scala/org/apache/daffodil/runtime2/TestCodeGenerator.scala

```diff
@@ -26,18 +26,31 @@ import org.apache.daffodil.compiler.Compiler
 import org.apache.daffodil.util.Misc
 import org.apache.daffodil.util.SchemaUtils
 import org.apache.daffodil.util.TestUtils
+import org.junit.After
 import org.junit.Assert.assertArrayEquals
+import org.junit.Before
 import org.junit.Test
 
 /**
  * Checks that we can create a [[CodeGenerator]] and call its methods.
  * The value of this test is to debug the call path from [[Compiler]]
- * to [[CodeGenerator]] for one very simple DFDL schema.  Running TDML
- * tests with daffodil-runtime2 is a more effective way to test the
- * functionality of CodeGenerator's generated code for as many DFDL
- * schemas as you could want.
+ * to [[CodeGenerator]] for a single test DFDL schema.  Running TDML
+ * tests with daffodil-runtime2 is a more effective way to check that
+ * CodeGenerator can generate appropriate code for as many DFDL schemas
+ * as you could want.
  */
 class TestCodeGenerator {
+  // Ensure all tests remove tmpOutputDir after using it
+  val tmpOutputDir: os.Path = os.pwd/"generate_tmp"
```

Comment:
Any particular reason to not use ``Files.createTempDirectory``? That has the advantage that it goes in /tmp, and ensures theat you have a unique, empty directory. Avoids the need for the before function to clean it out. I also noticed one of the windows tests failed in here somewhere with an error about this "generate_tmp" directory. I wonder if there is some kind of race condition here?

Reply:
Yes, I also noticed that one of the Windows builds failed because this directory couldn't be deleted, probably because both the daffodil-cli and daffodil-runtime2 tests were running at the same time and trying to use this directory at the same time.  Since I've been using os-lib calls for all my file operations, I've replaced the os.pwd/"generate_tmp" with calls to os.temp.dir() in the test methods.